src/modules/user/infraestructure/__test__/user-crud.controller.spec.ts

```diff
@@ -0,0 +1,93 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { UserController } from '../user-crud.controller';
+import { CreateUserDto, UpdateUserDto } from '../../application/dtos';
+
+describe('User Controller', () => {
+  let userController: UserController;
+
+  const mockUsers = [
+    {
+      id: 1,
+      nickName: 'John Doe',
+      email: 'john.doe@mail.com',
+      password: '123',
+    },
+    {
+      id: 2,
+      nickName: 'Jane Doe',
+      email: 'jane.doe@mail.com',
+      password: '123',
+    },
+  ];
+
+  const mockUserController = {
+    create: jest.fn((user: CreateUserDto) => {
+      const newUser = { ...user, id: mockUsers.length + 1 };
+      mockUsers.push(newUser);
+      return newUser;
+    }),
+
+    findAll: jest.fn(() => mockUsers),
+
+    findOne: jest.fn(
+      (id) => mockUsers.filter((u) => u.id.toString() === id)[0],
+    ),
+
+    update: jest.fn((id, user: UpdateUserDto) => {
+      const userFound = mockUsers.filter((u) => u.id === id)[0];
+      const updatedUser = { ...userFound, ...user };
+      mockUsers.splice(id - 1, 1, updatedUser);
+      return { updatedUser };
+    }),
+
+    remove: jest.fn((id) => {
+      const userFound = mockUsers.filter((u) => u.id.toString() === id)[0];
+      mockUsers.splice(id - 1, 1);
+      return userFound;
+    }),
```

Comment:
You have to mock your `userService`, not your controller. Also, you want just to mock it.
```
const yourMockService = {
  create: jest.fn(),
  findAll: jest.fn()
}
```